util/multisig/src/error.rs

```diff
@@ -0,0 +1,42 @@
+use failure::Context;
+
+#[derive(Debug)]
+pub struct Error {
+    inner: Context<ErrorKind>,
+}
+
+#[derive(Copy, Clone, Eq, PartialEq, Debug)]
+pub struct Threshold {
+    pub threshold: usize,
+    pub pass_sigs: usize,
+}
+
+#[derive(Copy, Clone, Eq, PartialEq, Debug, Fail)]
+pub enum ErrorKind {
+    #[fail(display = "The count of sigs should less than pks.")]
+    SigCountOverflow,
+    #[fail(display = "The count of sigs less than threshold.")]
+    SigNotEnough,
+    #[fail(display = "Failed to meet threshold {:?}.", _0)]
+    Threshold(Threshold),
```

Comment:
Enum supports struct like variant:

```suggestion
    Threshold {
        threshold: usize,
        pass_sigs: usize,
    },
```

Reply:
Ops didn't know this before